js/core/contrib.js

```diff
@@ -0,0 +1,98 @@
+                    msg.pub("end", "core/contrib");
+                    cb();
+                }
+                
+                function prop(prop) {
+                    return function (o) {
+                        return o[prop];
+                    };
+                }
+                
+                function slice(args) {
+                    return Array.prototype.slice.call(args, 0)
+                }
+
+                function findUsers() {
+                    var users = {};
+                    slice(arguments).forEach(function (things) {
+                        things.forEach(function (thing) {
+                            if (thing.user) {
+                                users[thing.user.url] = true;
+                            }
+                        });
+                    });
+                    return Object.keys(users);
+                }  
+
+                function join(things) {
+                    if (!things.length) {
+                        return "";
+                    }
+                    things = things.slice(0);
+                    var last = things.pop();
+                    var length = things.length;
+                    if (length === 0) {
+                        return last;
+                    }
+                    if (length === 1) {
+                        return things[0] + " and " + last;
+                    }
+                    return things.join(", ") + ", and " + last;
+                }
+
+                function toHTML(urls, editors, element) {
+                    return $.when.apply($, urls.map(function (url) {
+                        return github.fetch(url);
+                    })).then(function () {
+                        var names = slice(arguments).map(function (user) {
+                            user = user[0];
+                            return user.name || user.login;
+                        }).filter(function (name) {
+                            return editors.indexOf(name) < 0;
+                        });
+                        names.sort(function (a, b) {
+                            return a.toLowerCase().localeCompare(b.toLowerCase());
+                        });
+                        $(element).html(join(names)).attr("id", null);
+                    });
+                }
+
+                msg.pub("start", "core/contrib");
+                var $commenters = $(doc).find("#gh-commenters")[0];
+                var $contributors = $(doc).find("#gh-contributors")[0];
+                if (conf.githubAPI && ($commenters || $contributors)) {
+                    github.fetch(conf.githubAPI).then(function (json) {
+                        return $.when(
+                            github.fetchIndex(json.issues_url),
+                            github.fetchIndex(json.issue_comment_url),
+                            github.fetchIndex(json.contributors_url)
+                        ).then(function(issues, comments, contributors) {
```

Comment:
Whoa! The API gives you back more than one single value in the returned promise????